### PR TITLE
FEATURE: add and delete AssetCollections and Tags

### PR DESCRIPTION
**Pull request by dimaip**
Resolves: https://github.com/Flowpack/media-ui/issues/24

TODO:
- [x] get local cache to update
- [x] optimistic update?
- [x] create collection modal
- [x] create tag modal
Maybe later:
- [ ] expand collection when creating a tag
- [ ] setFocus doesn't work in modals for some reason

**What I did**

- Identify AssetCollections by id instead of title.
- Add and delete operations for AssetCollections
- Add and delete tags

**How to verify it**

Try to create and remove asset collections and tags. Make sure the state updates correctly, including optimistic update.

### CODE REVIEW

File: Classes/GraphQL/Resolver/Type/AssetCollectionResolver.php [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    public function id(AssetCollection $assetCollection): string
+    {
+        return $this->persistenceManager->getIdentifierByObject($assetCollection);
+    }
+
+    /**
+     * @param AssetCollection $assetCollection
+     * @return string
+     */
+    public function title(AssetCollection $assetCollection): string
```
[the remark this answers is not in the document]
Comment (dimaip) [reply Classes/GraphQL/Resolver/Type/AssetCollectionResolver.php:47]: Ah, nice!

File: Resources/Private/GraphQL/schema.root.graphql [hunk 2]
```diff
@@ -48,54 +48,42 @@ type Query {
...
-        assetSourceId: AssetSourceId!
-    ): Boolean!
+    deleteAsset(id: AssetId!, assetSourceId: AssetSourceId!): Boolean!
 
-    tagAsset(
-        id: AssetId!,
-        assetSourceId: AssetSourceId!,
-        tag: TagLabel!
-    ): Asset!
+    tagAsset(id: AssetId!, assetSourceId: AssetSourceId!, tag: TagLabel!): Asset!
```
Comment (Sebobo) [inline Resources/Private/GraphQL/schema.root.graphql:60]: Any reason why you changed all the formatting here?
The multilines were intentional.

Comment (dimaip) [reply Resources/Private/GraphQL/schema.root.graphql:60]: VSC/prettier or whatever else autoformats it.
Let's not do manual formatting, please, I hate overriding what VSC wants to do...

Comment (Sebobo) [reply Resources/Private/GraphQL/schema.root.graphql:60]: oh interesting, it didn't do that for me and I didn't know it applies to this filetype.

Comment (dimaip) [reply Resources/Private/GraphQL/schema.root.graphql:60]: I have graphql plugin installed in VSC, maybe it does it?

Comment (bwaidelich) [reply Resources/Private/GraphQL/schema.root.graphql:60]: and it didn't change that for the `updateAsset` mutation?

Comment (dimaip) [reply Resources/Private/GraphQL/schema.root.graphql:60]: apparently it'd go over the line length limit or something. I dunno, I just don't question what formatting the autoformat feature does for me usually...

Comment (dimaip) [reply Resources/Private/GraphQL/schema.root.graphql:60]: here's what it shows in the status bar
image

Comment (Sebobo) [reply Resources/Private/GraphQL/schema.root.graphql:60]: OK, I had to add graphql manually to the prettier config in PHPStorm, now it does the same 👍 
`updateAsset` is simply too long for one line.

**Comment by dimaip** [comment]
Ready for review, I'll do editing of collections/tags as a separate PR.
Also I didn't feel like changing tags to entities now, maybe I'll do it later.

**Comment by Sebobo** [comment]
Can you allow me to push on your branch?

I made some commits to fix some stuff and cleanup code.

**Review by Sebobo — CHANGES_REQUESTED** [review]
Found some issues but already fixed them, just need to push them onto the branch.
Then it's good to go :)

Just found out that removing collections from an asset doesn't properly work. Wasn't able to quickly fix it and will create a separate issue for that. Seems to be a problem on the php side.

**Comment by dimaip** [comment]
@Sebobo what do you mean give push access? It's a branch on your own repo...
Thanks so much for the review! ❤️

**Comment by Sebobo** [comment]
No the PR comes from your fork, and there I wasn't able to push to.

**Comment by dimaip** [comment]
No, it does not: 
image

**Comment by Sebobo** [comment]
Hm, when I checked it out it was still on your fork, I switched now. 🤷🏿‍♂️

**Review by Sebobo — APPROVED** [review]

**Comment by dimaip** [comment]
Thanks for the review and the fixes! Yay!

**Pull request merged**